src/osemgrep/reporting/Cli_json_output.ml

```diff
@@ -292,6 +292,61 @@ let cli_error_of_core_error (x : Out.core_error) : Out.cli_error =
 (*****************************************************************************)
 (* LATER: we should get rid of those intermediate Out.core_xxx *)
 
+(* This is a cursed function that calculates everything but the index part
+ * of the match_based_id. It's cursed because we need hashes to be exactly
+ * the same, but the algorithm used on the python side to generate
+ * the final string thats hashed has some python specific quirks.
```

Comment:
Ah! I told you :)

Reply:
Still I don't understand the use of "cursed" here.

Reply:
Yea, I forgot about the tuple stuff

Reply:
This function is cursed because it shouldn't exist, i.e. in a perfect world, we wouldn't need to emulate python things in ocaml.